packages/flutter/lib/src/material/button.dart

```diff
@@ -325,7 +326,16 @@ class _RenderInputPadding extends RenderShiftedBox {
 
   @override
   bool hitTest(HitTestResult result, { Offset position }) {
-    return super.hitTest(result, position: position) ||
-      child.hitTest(result, position: child.size.center(Offset.zero));
+    if (super.hitTest(result, position: position)) {
+      return true;
+    }
+    final Offset center = child.size.center(Offset.zero);
+    final Matrix4 transform = Matrix4.identity()
+      ..setRow(0, Vector4(0, 0, 0, center.dx))
+      ..setRow(0, Vector4(0, 0, 0, center.dy));
+    result.pushTransform(transform);
```

Comment:
instead of working out the transform, don't worry about it. `result.disableTransform()` or `result.forceToPoint(offset)` or something.